lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ServiceAndContextConfigurationProvider.cs

```diff
@@ -69,113 +66,245 @@ internal sealed class ServiceAndContextConfigurationProvider<TMapInput, TMapOutp
         /// submitted evaluator to failed evaluator
         /// </summary>
         /// <param name="evaluatorId"></param>
-        internal void EvaluatorFailed(string evaluatorId)
+        /// <returns>Whether failed evaluator is master or not</returns>
+        internal bool EvaluatorFailed(string evaluatorId)
```

Comment:
This needs a better name. In general, I prefer methods with a verb starting the name. In this case, I'd suggest to: `RecordFailedEvaluatorById()` and `IsMasterEvaluatorId()`.